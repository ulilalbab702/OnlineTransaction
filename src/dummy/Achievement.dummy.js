export const Achievement = [
    {
        "Code": "FM-JBYPM",
        "Name": "FMC Jembayan PAMA",
        "Percent": 50,
        "BackLogOpen": 1,
        "BackLogClose": 1,
        "Priority1": 1,
        "Priority2": 0,
        "Priority3": 1,
        "Priority4": 0
    }
]