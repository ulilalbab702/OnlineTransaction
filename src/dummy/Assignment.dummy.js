export const listOfMechanicDummy = [
    {
      "Username": "string",
      "FirstName": "string",
      "LastName": "string",
      "Email": "string",
      "MechanicUT": true,
      "Jobs": [
        {
          "JobType": "string",
          "Total": 0
        }
      ]
    }
  ]
export const jobSummaryDummy = {
  "Date": "2020-06-04T16:18:05.5472055",
  "CreatedBy": "AGUS LISANTO",
  "AssignedBy": "AGUS LISANTO",
  "MechanicLeader": {
      "UserName": "82107038@unitedtractors.com",
      "Name": "SUPRIYADI SUPRIYADI"
  },
  "Mechanics": [],
  "HandoverDate": null,
  "HandoverCreatedBy": null,
  "HandoverAssignedBy": null,
  "HandoverMechanicLeader": null,
  "HandoverMechanics": null,
  "Job": {
      "WoId": "8ff577d1-2911-4bbd-9d55-ffad4fe76fc5",
      "WoNumber": "71018650",
      "JobType": "INS",
      "UnitModel": "HD465-7",
      "SerialNumber": "15163",
      "UnitCode": "HT3022",
      "EquipmentNumber": "71506",
      "IsHandover": false,
      "Id": "a162bd06-6aca-499a-9f97-a5839b8a0612",
      "CreatedBy": "80107002@unitedtractors.com",
      "CreatedDate": "2020-06-04T16:18:05.5472055",
      "UpdatedBy": "80107002@unitedtractors.com",
      "UpdatedDate": "2020-06-04T16:18:05.5472055",
      "IsActive": true
  }
}