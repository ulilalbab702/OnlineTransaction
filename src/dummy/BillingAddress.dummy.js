export const dataBillingAddressDummy = [
    {
        "fullName": "Test Integrasi",
        "contactNumber": "088122345",
        "email": "testintegrasi@unitedtractors.com",
        "position": "string",
        "addressLabel": "Kantor PT Global Service Indonesia",
        "city": "Jakarta",
        "address": "Jl Raya kaliabang Tengah, Ruko San Jaya, Blok A5 No 9,\r\nSebalah Indomaret",
        "postalCode": "13920",
        "locationLatitude": "3.669276961596295",
        "locationLongitude": "102.4003031491451",
        "isPrimary": true
    }]