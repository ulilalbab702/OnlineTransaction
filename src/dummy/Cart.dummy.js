export const cartDummyList = {
    "cartProductResponse": {
        "data": [
            {
                "cartId": "d7e23521-0d23-4f37-aceb-77f8f75b3d16",
                "productId": "32bdd633-932f-465a-b93c-219a70971008",
                "name": "600-311-8291",
                "imageThumbnailUrl": "https://untr-my.sharepoint.com/:i:/g/personal/cst_dev04_unitedtractors_com/EW__sgepd2xJs1m7uxNquHQBbRKmqLZ3DkmR7N1Mb7JWww?e=1f9bRa",
                "qty": 3,
                "priceAfter": 300000,
                "priceBefore": 500000,
                "currency": "IDR",
                "cookiesData": ""
            },
            {
                "cartId": "10d112f9-0a0b-4c26-87b6-87ffb0374a77",
                "productId": "4211034d-8571-4ed6-84b2-8b380443bf7b",
                "name": "568002",
                "imageThumbnailUrl": "https://untr-my.sharepoint.com/:i:/g/personal/cst_dev04_unitedtractors_com/EW__sgepd2xJs1m7uxNquHQBbRKmqLZ3DkmR7N1Mb7JWww?e=1f9bRa",
                "qty": 2,
                "priceAfter": 800000,
                "priceBefore": 1000000,
                "currency": "IDR",
                "cookiesData": ""
            }
        ],
        "meta": {
            "totalItems": 2,
            "pageNumber": 1,
            "pageSize": 5,
            "totalPages": 1,
            "hasPreviousPage": false,
            "hasNextPage": false,
            "nextPageNumber": 1,
            "previousPageNumber": 1
        }
    },
    "subTotalPrice": 2500000
}
