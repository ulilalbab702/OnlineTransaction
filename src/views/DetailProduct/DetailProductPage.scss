body {
  background-color: #ffffff;
}

.bgTP {
  margin-left: 150px;
}

@media only screen and (min-width: 1280px) {
  .containerTP {
    margin-top: 0.5rem;
  }

  .marginTP {
    margin-top: 6rem;
    width: 100%;
  }

  .marginTP1 {
    margin-top: 1.5rem;
    margin-bottom: 1.5rem;
    display: flex;
    width: 100%;
  }

  .marginButtonTP {
    margin-left: 2rem;
  }

  .styleTP {
    font-family: "SFUIText-Regular";
    font-size: 20px;
    line-height: 24px;
    color: #3c3b3b;
    font-weight: 600;
  }

  .TPord {
    float: left;
    width: 93%;
  }

  .imageDetailProduct {
    margin-top: 10px;
    margin-bottom: 10px;
    width: 350px;
    height: 296px;
    padding: 20px;
    margin: 50px;
  }

  .titleProductDetail {
    font-family: "SF Pro Text";
    font-style: normal;
    font-weight: bold;
    font-size: 28px;
    line-height: 20px;
    letter-spacing: 0.02em;
    color: #000000;
  }

  .priceProduct {
    font-family: "SF Pro Text";
    font-style: normal;
    font-weight: bold;
    font-size: 28px;
    line-height: 20px;
    color: #148f0f;
    margin-left: 15px;
  }

  .priceBefore {
    font-style: normal;
    font-weight: normal;
    font-size: 14px;
    line-height: 12px;
    color: #000000;
    opacity: 0.75;
    text-decoration-line: line-through;
    text-decoration-style: solid;
    margin-left: 30px;
    margin-top: 5px;
  }

  .discountProduct {
    font-style: normal;
    font-weight: normal;
    font-size: 12px;
    line-height: 12px;
    color: #c60242;
    margin-left: 20px;
    background-color: #ffacc7;
    border-radius: 10px;
    padding-top: 5px;
    padding-bottom: 5px;
    padding-left: 10px;
    padding-right: 10px;
  }

  .titleRating {
    font-family: "SF Pro Text";
    font-style: normal;
    font-weight: bold;
    font-size: 20px;
    line-height: 20px;
    letter-spacing: 0.02em;
    color: #000000;
    margin-left: 20px;
  }

  .descRating {
    font-family: "SF Pro Text";
    font-style: normal;
    font-size: 20px;
    line-height: 20px;
    letter-spacing: 0.02em;
    color: #000000;
    margin-left: 20px;
    opacity: 0.5;
  }

  .titleBrandProduct {
    font-family: "SF Pro Text";
    font-style: normal;
    font-weight: normal;
    font-size: 22px;
    line-height: 20px;
    color: #000000;
    opacity: 0.4;
  }

  .ratingProduct {
    margin-left: 15px;
    height: 20px;
    width: 20px;
  }

  .serialNumberProduct {
    flex: 0.5;
    font-family: "SF Pro Text";
    font-style: normal;
    font-weight: normal;
    font-size: 18px;
    line-height: 21px;
    color: #404040;
    opacity: 0.8;
    margin-left: 15px;
  }
  .unitCodeProduct {
    font-family: "SF Pro Text";
    font-style: normal;
    font-weight: 600;
    font-size: 18px;
    line-height: 21px;
    color: #404040;
    opacity: 0.8;
    margin-left: 5px;
  }
  .canBeUse {
    font-family: "SF Pro Text";
    font-style: normal;
    font-weight: normal;
    font-size: 18px;
    line-height: 21px;
    color: #404040;
    opacity: 0.8;
    margin-left: 15px;
  }
  .unitCodeBrand {
    font-family: "SF Pro Text";
    font-style: normal;
    font-weight: 600;
    font-size: 14px;
    line-height: 17px;
    text-align: center;
    color: #000000;
    background: #f2f2f2;
    border: 1px solid #b3b3b3;
    box-sizing: border-box;
    border-radius: 8px;
    padding-top: 5px;
    padding-bottom: 5px;
    padding-left: 10px;
    padding-right: 15px;
    margin-left: 5px;
  }
  .unitDiscountProduct {
    font-family: "SF Pro Text";
    font-style: normal;
    font-weight: 600;
    font-size: 16px;
    line-height: 19px;
    color: #000000;
    opacity: 0.8;
    background: #ffd500;
    padding: 5px;
    display: inline-block;
  }
  .buyButton {
    padding-top: 15px;
    width: 204.96px;
    height: 50.86px;
    border: 1px solid #000000;
    box-sizing: border-box;
    border-radius: 8px;
    font-family: "SF Pro Text";
    font-style: normal;
    font-weight: 600;
    font-size: 14px;
    text-align: center;
    color: #000000;
  }
  .addButton {
    margin-left: 20px;
    padding-top: 15px;
    width: 318.13px;
    height: 50.86px;
    background: #ffd500;
    border-radius: 8px;
    font-family: "SF Pro Text";
    font-style: normal;
    font-weight: 600;
    font-size: 14px;
    line-height: 17px;
    text-align: center;
    color: #000000;
  }
  .descriptionFavourite {
    font-family: "SF Pro Text";
    font-style: normal;
    font-weight: normal;
    font-size: 22px;
    line-height: 26px;
    color: #404040;
    margin-left: 5rem;
    opacity: 0.8;
  }
  .relateProduct {
    font-family: "SF Pro Text";
    font-style: normal;
    font-weight: bold;
    font-size: 22px;
    line-height: 26px;
    color: #000000;
    margin-left: 7rem;
    margin-top: 2rem;
  }
  .descriptionProduct {
    font-family: "SF Pro Text";
    font-style: normal;
    font-weight: bold;
    font-size: 22px;
    line-height: 26px;
    color: #404040;
    margin-left: 5rem;
    opacity: 0.8;
    cursor: pointer;
  }
  .descriptionProductActive {
    font-family: "SF Pro Text";
    font-style: normal;
    font-weight: bold;
    font-size: 22px;
    line-height: 26px;
    color: #000000;
    margin-left: 5rem;
    border-spacing: 5px;
    border-bottom: 5px solid #ffd500;
    margin-bottom: 0px;
    cursor: pointer;
  }
  .discountDetail {
    width: 25px;
    height: 29px;
  }
  .iconFavourite {
    width: 25px;
    height: 25px;
    background-color: #f1f1f1;
    padding: 5px;
    border-radius: 50px;
    margin-left: 10px;
  }
  .gridListRelate {
    display: grid;
    grid-template-columns: repeat(6, 1fr);
    grid-auto-rows: minmax(5px, auto);
    grid-gap: 5px;
  }
  .gridListImage {
    display: grid;
    grid-template-columns: repeat(4, 1fr);
    grid-auto-rows: minmax(5px, auto);
    grid-gap: 5px;
  }
  .imageListProduct {
    transform: scale(0.9) !important;
    max-width: 100%;
    max-height: 100%;
    margin-left: -30px;
  }
  .imageItem {
    transform: scale(1) !important;
    max-width: 100%;
    max-height: 90%;
  }
  .cardImageItem {
    display: flex;
    border-radius: 8px;
    width: 80px;
    margin-top: 20px;
    margin-bottom: 10px;
    background-color: #f1f1f1;
    height: 80px;
  }
  .iconAva {
    margin-left: 8rem;
    width: 70px;
    height: 70px;
    border-radius: 50px;
  }
  .titleName {
    font-family: "SF UI Text";
    font-style: normal;
    font-weight: 600;
    font-size: 20px;
    line-height: 24px;
    color: #515151;
  }
  .ratingReview {
    margin-left: 5px;
    height: 20px;
    width: 20px;
  }
  .dateRating {
    font-family: SF UI Text;
    font-style: normal;
    font-weight: 300;
    font-size: 14px;
    line-height: 17px;
    color: #b4b4b4;
    margin-top: 5px;
  }
  .reviewDesc {
    font-family: SF UI Text;
    font-style: normal;
    font-weight: 300;
    font-size: 16px;
    line-height: 150%;
    letter-spacing: 0.5px;
    opacity: 0.7;
  }
}

@media only screen and (min-width: 992px) and (max-width: 1279px) {
  .bgTP {
    margin-left: 150px;
  }
  .containerTP {
    margin-top: 0.5rem;
  }

  .marginTP {
    margin-top: 6rem;
    width: 100%;
  }

  .marginTP1 {
    margin-top: 1.5rem;
    margin-bottom: 1.5rem;
    display: flex;
    width: 100%;
  }

  .marginButtonTP {
    margin-left: 2rem;
  }

  .styleTP {
    font-family: "SFUIText-Regular";
    font-size: 20px;
    line-height: 24px;
    color: #3c3b3b;
    font-weight: 600;
  }

  .TPord {
    float: left;
    width: 93%;
  }
}

.listProductItem {
  width: 400px;
}
