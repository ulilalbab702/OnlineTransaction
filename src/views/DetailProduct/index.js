import DetailProductPage from './DetailProductPage';
import { connect } from 'react-redux';
import {
    BRAND_REQ,
    CATEGORY_REQ,
    FORGOT_PASSWORD,
    IMAGE_SLIDER_REQ,
    SIGNUP,
    VERIFY_CODE
} from "actions/actionTypes";
import { LOGIN_TYPE } from 'actions/actionTypes';
import { forgotPassword, login, signup, verifyCode, changePassword, loginGoogle } from 'actions/user';
import { push } from 'connected-react-router';
import {
    createLoadingSelector,
} from 'utils/selector.helper'
import { detailProductGet, detailProductRelatedGet, reviewDetailProduct } from 'actions/detailProduct';
import { clearError, clearStore } from 'actions/category';
import {
    blockLogin,
    countDownTimerLock,
    countFailedLogin,
    releaseBlockedLogin,
    initialBlockedLoginStatus
} from 'actions/blockedLogin';
import { addWishlist, deleteWishlist } from 'actions/wishlist';
import { postCartData } from 'actions/listNews';
import { searchByUserId } from 'actions/cart';
import { termConditionGet, updateTermCondition } from 'actions/termCondition';

//Set selector loading 
const loadingSelector = createLoadingSelector([LOGIN_TYPE, SIGNUP, FORGOT_PASSWORD, VERIFY_CODE]);
const mapStateToProps = state => ({
    ...state.user,
    isLoading: loadingSelector(state),
    detailProduct: state.detailProduct.detailProduct.data,
    reviewProductData: state.detailProduct.detailProduct.dataReview,
    loadingCart: state.listNews.listNews.loadingCart,
    errorCartData: state.listNews.listNews.errorCartData,
    cartData: state.listNews.listNews.cartData,
    
    register: state.user.register,
    errorRegister: state.user.errorRegister,
    errorStatus: state.category.errorStatus,
    error: state.error,
    errorForgot: state.error.errorForgot,
    errorVerify: state.error.errorVerify,
    verify: state.user.verify,
    dataTermCondition: state.termCondition.termCondition.data,

    failed_count: state.blockedLogin.blockedLogin.failed_count,
    lock_login: state.blockedLogin.blockedLogin.lock_login,
    timer_lock: state.blockedLogin.blockedLogin.timer_lock
});

const mapDispatchToProps = dispatch => ({
    push: (url, item) => dispatch(push(url, item)),
    fetchDetailProduct: (id) => dispatch(detailProductGet(id)),
    fetchCartPost: (token, body) => dispatch(postCartData(token, body)),
    fetchCartByUserId: (userId, token) => dispatch(searchByUserId(userId, token)),
    fetchReviewProduct: (productId) => dispatch(reviewDetailProduct(productId)),
    fetchAddWishlist: (token, data) => dispatch(addWishlist(token, data)),
    fetchDeleteSigleWishlist: (token, data) => dispatch(deleteWishlist(token, data)),
    fetchTermCondition: (token) => dispatch(termConditionGet(token)),
    fetchUpdateTermCondition: (token, version) => dispatch(updateTermCondition(token, version)),

    login: (data) => dispatch(login(data)),
    signup: (data) => dispatch(signup(data)),
    clearError: () => dispatch(clearError()),
    clearStore: () => dispatch(clearStore()),
    forgotPassword: ({ email, applicationId }) => dispatch(forgotPassword({ email, applicationId })),
    verifyCode: (data) => dispatch(verifyCode(data)),
    changePassword: (data) => dispatch(changePassword(data)),
    loginGoogle: (data) => dispatch(loginGoogle(data)),

    blockLogin: () => dispatch(blockLogin()),
    countDownTimerLock: (timer) => dispatch(countDownTimerLock(timer)),
    countFailedLogin: (failed_login) => dispatch(countFailedLogin(failed_login)),
    releaseBlockedLogin: () => dispatch(releaseBlockedLogin()),
    initialBlockedLoginStatus: () => dispatch(initialBlockedLoginStatus())
});

export default connect(mapStateToProps, mapDispatchToProps)(DetailProductPage);