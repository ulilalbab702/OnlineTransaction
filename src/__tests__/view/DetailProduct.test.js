import React from "react";
import toJson from "enzyme-to-json";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import DetailProductPage from "views/DetailProduct/DetailProductPage";
import { listDetailProductDummy, dataDetailProductDummy } from "./../../dummy/DetailProduct.dummy";

Enzyme.configure({ adapter: new Adapter() });
describe("Detail Product Component", () => {
	const wrapper = shallow(
		<DetailProductPage 
            detailProduct={dataDetailProductDummy}
			detailProductRelated={listDetailProductDummy}
		/>
	);
	describe("Detail Product Component", () => {
		it("Should match to snapshot", () => {
			expect(toJson(wrapper)).toMatchSnapshot();
		});
	});
});
