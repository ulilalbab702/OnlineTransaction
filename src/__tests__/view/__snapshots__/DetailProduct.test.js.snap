// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Product Component Detail Product Component Should match to snapshot 1`] = `
<div
  style={
    Object {
      "marginBottom": "4rem",
    }
  }
>
  <Row
    className="defaultFlexProduct"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <div
      className="bgTP"
      style={
        Object {
          "paddingLeft": "0",
        }
      }
    >
      <p
        className="titleBrandProduct"
      />
      <p
        className="titleProductDetail"
      />
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <img
          className="ratingProduct"
          src="Star.png"
        />
        <p
          className="titleRating"
        >
          5.0
        </p>
        <p
          className="descRating"
        >
          (undefined Review)
        </p>
        <p
          className="titleRating"
        >
          â€¢
        </p>
        <p
          className="descRating"
        >
          Sold
        </p>
        <p
          className="titleRating"
        >
          56x
        </p>
      </Row>
      <hr
        className="mt-0"
      />
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <p
          className="priceProduct"
        >
          IDR
           
        </p>
        <p
          className="priceBefore"
        />
        <p
          className="discountProduct"
        >
          25%
        </p>
      </Row>
      <hr
        className="mt-0"
      />
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <p
          className="serialNumberProduct"
        >
          Serial Number
        </p>
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Row
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <p
              className="unitCodeProduct"
            >
              100801
            </p>
            <p
              className="canBeUse"
            >
              (can be used for unit Model)
            </p>
          </Row>
          <Row
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <p
              className="unitCodeBrand"
            >
              PC200
            </p>
            <p
              className="unitCodeBrand"
            >
              PC210
            </p>
            <p
              className="unitCodeBrand"
            >
              PC195
            </p>
            <p
              className="unitCodeBrand"
            >
              PC2000
            </p>
          </Row>
        </Col>
      </Row>
      <hr
        className="mt-0"
      />
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <p
          className="serialNumberProduct"
        >
          Promotion
        </p>
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Row
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <p
              className="unitDiscountProduct"
            >
              Diskon 30%, Hingga 2.000.000
            </p>
            <img
              className="discountDetail"
              src="discountDetail.png"
            />
          </Row>
          <Row
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <p
              className="unitDiscountProduct"
            >
              Buy 2, Diskon Hingga 30%
            </p>
            <img
              className="discountDetail"
              src="discountDetail.png"
            />
          </Row>
        </Col>
      </Row>
      <hr
        className="mt-0"
      />
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <p
          className="serialNumberProduct"
        >
          Kode Product
        </p>
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <p
            className="serialNumberProduct"
          >
            600-311-8293
          </p>
        </Col>
      </Row>
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <p
          className="serialNumberProduct"
        >
          Status
        </p>
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <p
            className="serialNumberProduct"
          >
            Tersedia
          </p>
        </Col>
      </Row>
      <hr
        className="mt-0"
      />
      <Row
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <p
          className="buyButton"
        >
          Buy Now
        </p>
        <p
          className="addButton"
        >
          Add Bucket
        </p>
      </Row>
    </div>
  </Row>
  <hr
    className="mt-0"
  />
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Row
      style={
        Object {
          "flex": 1,
          "marginLeft": "1rem",
        }
      }
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <p
        className="descriptionProductActive"
        onClick={[Function]}
      >
        Description
      </p>
      <p
        className="descriptionProduct"
        onClick={[Function]}
      >
        Review (2)
      </p>
    </Row>
    <Row
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
          "marginRight": "4rem",
        }
      }
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <p
        className="descriptionFavourite"
      >
        Favourite
      </p>
      <img
        className="iconFavourite"
        src="heart.png"
      />
      <p
        className="descriptionFavourite"
      >
        Share
      </p>
      <img
        className="iconFavourite"
        src="share.png"
      />
    </Row>
  </Row>
  <hr
    className="mt-0"
  />
  <Col
    style={
      Object {
        "marginLeft": "3rem",
      }
    }
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <p
      style={
        Object {
          "marginLeft": "3rem",
          "textAlign": "justify",
          "width": "60%",
        }
      }
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
    </p>
  </Col>
  <p
    className="relateProduct"
  >
    Related Product
  </p>
  <Row
    className="gridListRelate  defaultMargin"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <div
      className="cardList"
    >
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <div
          className="cardImage"
        >
          <img
            className="imageList"
            src="https://firebasestorage.googleapis.com/v0/b/utportal-d20a4.appspot.com/o/Images%2F37978014-625e-41ff-b29f-639ce3451d42%5C.png?alt=media&token=35f346eb-3199-41e8-a6c6-1fad0a5ee73d"
          />
        </div>
        <p
          className="titleKategoriProduct"
        >
          Komatsu
        </p>
        <p
          className="titleProduct"
        >
          Cartridge
        </p>
        <p
          className="titlePrice"
        />
        <p
          className="titlePriceDiscount"
        />
        <p
          className="titleKategoriProduct"
        >
          <img
            src="Star.png"
            style={
              Object {
                "height": 10,
                "width": 10,
              }
            }
          />
           ( 
           review )
        </p>
      </Col>
    </div>
    <div
      className="cardList"
    >
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <div
          className="cardImage"
        >
          <img
            className="imageList"
            src="https://firebasestorage.googleapis.com/v0/b/utportal-d20a4.appspot.com/o/Images%2Fcc9a0189-d1ab-4d8e-a304-8c75ee88fdc6%5C.png?alt=media&token=5fdf1c91-688b-497b-9d2e-6f7a58d0b905"
          />
        </div>
        <p
          className="titleKategoriProduct"
        >
          Komatsu
        </p>
        <p
          className="titleProduct"
        >
          Cartridge
        </p>
        <p
          className="titlePrice"
        />
        <p
          className="titlePriceDiscount"
        />
        <p
          className="titleKategoriProduct"
        >
          <img
            src="Star.png"
            style={
              Object {
                "height": 10,
                "width": 10,
              }
            }
          />
           ( 
           review )
        </p>
      </Col>
    </div>
    <div
      className="cardList"
    >
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <div
          className="cardImage"
        >
          <img
            className="imageList"
            src="https://firebasestorage.googleapis.com/v0/b/utportal-d20a4.appspot.com/o/Images%2F7e2fa93a-4485-45af-ba90-e3a5a08333da%5C.png?alt=media&token=448bae01-fac7-4d55-ba87-22d051c5e6b8"
          />
        </div>
        <p
          className="titleKategoriProduct"
        >
          Bomag
        </p>
        <p
          className="titleProduct"
        >
          Cartridge
        </p>
        <p
          className="titlePrice"
        />
        <p
          className="titlePriceDiscount"
        />
        <p
          className="titleKategoriProduct"
        >
          <img
            src="Star.png"
            style={
              Object {
                "height": 10,
                "width": 10,
              }
            }
          />
           ( 
           review )
        </p>
      </Col>
    </div>
    <div
      className="cardList"
    >
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <div
          className="cardImage"
        >
          <img
            className="imageList"
            src="https://firebasestorage.googleapis.com/v0/b/utportal-d20a4.appspot.com/o/Images%2F84921cd8-fcd7-4c1e-816f-6a7d3b9868e6%5C.png?alt=media&token=f66c6b33-cfcb-462e-b8b5-7248af4fbdf1"
          />
        </div>
        <p
          className="titleKategoriProduct"
        >
          Tadano
        </p>
        <p
          className="titleProduct"
        >
          Cartridge
        </p>
        <p
          className="titlePrice"
        />
        <p
          className="titlePriceDiscount"
        />
        <p
          className="titleKategoriProduct"
        >
          <img
            src="Star.png"
            style={
              Object {
                "height": 10,
                "width": 10,
              }
            }
          />
           ( 
           review )
        </p>
      </Col>
    </div>
    <div
      className="cardList"
    >
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <div
          className="cardImage"
        >
          <img
            className="imageList"
            src="https://firebasestorage.googleapis.com/v0/b/utportal-d20a4.appspot.com/o/Images%2F116364fd-0839-4fef-9d2e-bc22d39cd3df%5C.png?alt=media&token=a1cc5faa-5c49-4da9-accc-ff2cfd48f041"
          />
        </div>
        <p
          className="titleKategoriProduct"
        >
          UD Trucks
        </p>
        <p
          className="titleProduct"
        >
          Cartridge
        </p>
        <p
          className="titlePrice"
        />
        <p
          className="titlePriceDiscount"
        />
        <p
          className="titleKategoriProduct"
        >
          <img
            src="Star.png"
            style={
              Object {
                "height": 10,
                "width": 10,
              }
            }
          />
           ( 
           review )
        </p>
      </Col>
    </div>
  </Row>
</div>
`;
